packages/amplify-provider-awscloudformation/src/aws-utils/aws-cfn.js

```diff
@@ -177,8 +179,17 @@ class CloudFormation {
    */
   filterFailedStackEvents(eventsWithFailure) {
     return eventsWithFailure
-      .filter(stack => stack.ResourceType !== 'AWS::CloudFormation::Stack')
-      .filter(stack => this.eventMap['eventToCategories'].has(stack.LogicalResourceId))
+      .filter(
+        stack =>
+          stack.ResourceType !== 'AWS::CloudFormation::Stack' ||
+          (stack.ResourceStatusReason && !stack.ResourceStatusReason.includes(STACK_RESOURCE_FILTER_FAIL_REASON)),
+      )
+      .filter(
+        stack =>
+          (this.eventMap['eventToCategories'] && this.eventMap['eventToCategories'].has(stack.LogicalResourceId)) ||
+          (this.eventMap['rootResources'] &&
+            this.eventMap['rootResources'].map(resource => resource.key).includes(stack.LogicalResourceId)),
```

Comment:
Nit: It's not necessary to map over the whole array, then loop over it again with includes. Using `.some` with `.includes` on the resource key string should be sufficient. Not a blocker.
```suggestion
            this.eventMap['rootResources'].some(resource => resource.key?.includes(stack.LogicalResourceId))),
```

Reply:
Good point, should we do `equal` check then? 
```suggestion
            this.eventMap['rootResources'].some(resource => resource.key?.equals(stack.LogicalResourceId)),
```